python/tk_maya/panel_generation.py

```diff
@@ -108,6 +109,20 @@ def dock_panel(engine, shotgun_panel, title, new_panel):
             # Embed the Shotgun app panel into the Maya panel workspace control.
             build_workspace_control_ui(shotgun_panel_name)
 
+            # Use a workaround to force Maya 2017 to refresh the panel size.
```

Comment:
i think it could be helpful to add some debug statements here so that we can keep track of what is going on